options/posix/generic/arpa-inet-stubs.cpp

```diff
@@ -146,8 +146,32 @@ const char *inet_ntop(int af, const void *__restrict src, char *__restrict dst,
 	errno = ENOSPC;
 	return NULL;
 }
-int inet_pton(int, const char *__restrict, void *__restrict) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+int inet_pton(int af, const char *__restrict src, void *__restrict dst) {
+	switch (af) {
+		case AF_INET: {
+			auto addr = reinterpret_cast<struct in_addr*>(dst);
+			auto dest = reinterpret_cast<uint8_t*>(&addr->s_addr);
```

Comment:
Why is this cast needed?

Reply:
``s_addr`` is of type ``in_addr_t`` (aka. ``uint32_t``) so i cast it to a byte array instead.

Reply:
That's quite ugly. We should assemble the uint32_t locally or declare a local `uint8_t` array that we memcpy to `s_addr`.